Lib/ufo2ft/kernFeatureWriter.py

```diff
@@ -232,3 +323,77 @@ def _getGlyphKerning(self, glyphName, i=None):
             if (glyphName in pair) if i is None else (pair[i] == glyphName):
                 hits.append((pair, value))
         return hits
+
+    def _scriptIsRtl(self, script):
+        """Return whether a script is right-to-left for kerning purposes.
+
+        References:
+        https://github.com/Tarobish/Jomhuria/blob/a21c41453ea8e3893e003ae9d5bee9ba7ac42d77/tools/getKernFeatureFromUFO.py#L18
+        https://github.com/behdad/harfbuzz/blob/691086f131cb6c9d97e98730c27673484bf93f87/src/hb-common.cc#L446
+        http://unicode.org/iso15924/iso15924-codes.html
+        """
+
+        return script in (
+            # Unicode-1.1 additions
+            'arab',  # ARABIC
+            'hebr',  # HEBREW
+
+            # Unicode-3.0 additions
+            'syrc',  # SYRIAC
+            'thaa',  # THAANA
+
+            # Unicode-4.0 additions
+            'cprt',  # CYPRIOT
+
+            # Unicode-4.1 additions
+            'khar',  # KHAROSHTHI
+
+            # Unicode-5.0 additions
+            'phnx',  # PHOENICIAN
+            'nkoo',  # NKO
+
+            # Unicode-5.1 additions
+            'lydi',  # LYDIAN
+
+            # Unicode-5.2 additions
+            'avst',  # AVESTAN
+            'armi',  # IMPERIAL_ARAMAIC
+            'phli',  # INSCRIPTIONAL_PAHLAVI
+            'prti',  # INSCRIPTIONAL_PARTHIAN
+            'sarb',  # OLD_SOUTH_ARABIAN
+            'orkh',  # OLD_TURKIC
+            'samr',  # SAMARITAN
+
+            # Unicode-6.0 additions
+            'mand',  # MANDAIC
+
+            # Unicode-6.1 additions
+            'merc',  # MEROITIC_CURSIVE
+            'mero',  # MEROITIC_HIEROGLYPHS
+
+            # Unicode-7.0 additions
+            'mani',  # MANICHAEAN
+            'mend',  # MENDE_KIKAKUI
+            'nbat',  # NABATAEAN
+            'narb',  # OLD_NORTH_ARABIAN
+            'palm',  # PALMYRENE
+            'phlp',  # PSALTER_PAHLAVI
+
+            # Unicode-8.0 additions
+            'hung',  # OLD_HUNGARIAN
+
+            # Unicode-9.0 additions
+            'adlm',  # ADLAM
+            )
+
+    def _glyphIsRtl(self, name):
+        """Return whether the closest-associated unicode character is RTL."""
+
+        delims = ('.', '_')
+        uv = self.font[name].unicode
+        while uv is None and any(d in name for d in delims):
+            name = name[:max(name.rfind(d) for d in delims)]
+            uv = self.font[name].unicode
```

Comment:
The `name` may not be a glyph name in the font in every iteration, raising a key value error when trying to get the unicode value.